kotlinx-coroutines-core/common/src/CancellableContinuationImpl.kt

```diff
@@ -165,8 +173,18 @@ internal open class CancellableContinuationImpl<in T>(
         return getSuccessfulResult(state)
     }
 
-    override fun resumeWith(result: Result<T>) =
+    override fun resumeWith(result: Result<T>) {
         resumeImpl(result.toState(), resumeMode)
+    }
+
+    override fun resume(value: T, onCancellation: CompletionHandler) {
+        if (!resumeImpl(CompletedWithCancellation(value, onCancellation), resumeMode)) {
+            // too late to resume (was cancelled) -- call handler
+            invokeHandlerSafely {
+                onCancellation.invokeIt((state as? CompletedExceptionally)?.cause)
```

Comment:
`null` shouldn't be allowed here, it is better to use `!!` or explicit invariant check than safe call